docs/2 - quickstart.md

```diff
@@ -0,0 +1,260 @@
+];
+```
+
+### Assign Permissions to a Group
+
+Each group can have its own specific set of permissions. These are defined in `Config\AuthGroups::$matrix`. You can specify each permission by it's full name, or using the context and an asterisk (*) to specify all permissions within that context.
+
+```php
+public array $matrix = [
+    'superadmin' => [
+        'admin.*',
+        'users.*',
+        'beta.access',
+    ],
+    //
+];
+```
+
+
+### Assign Permissions to a User
+
+Permissions can also be assigned directly to a user, irregardless of what groups they are a part of. This is done programatically on the `User` Entity.
+
+```php
+$user = auth()->user();
+
+$user->addPermission('users.create', 'beta.access');
+```
+
+This will add all new permissions. You can also sync permissions so that the user ONLY has the given permissions directly assigned to them. Any not in the provided list are removed from the user.
+
+```php
+$user = auth()->user();
+
+$user->syncPermissions(['users.create', 'beta.access']);
+```
+
+## Check If a User Has Permission
+
+When you need to check if a user has a specific permission, you want to check both within the groups they are a part of, as well as any specifically assigned to them, you should use the `can()` method on the `User` entity.
+
+```php
+if (! auth()->user()->can('users.create')) {
+    return redirect()->back()->with('error', 'You do not have permissions to access that page.');
+}
+```
+
+This can also be done through a [controller filter](https://codeigniter.com/user_guide/incoming/filters.html) if you want to apply it to multiple pages of your site.
+
+
+
+
+
+## Managing Users
+
+Since Shield uses a more complex user setup than many other systems, due to the [User Identities](1-concepts.md#identities), this quick overview should help you feel more confident when working with users on a day-to-day basis.
+
+### Creating Users
+
+By default, the only values stored in a user is the username. The first step is to create the user record with the username. If you don't have a username, be sure to set the value to `null` anyway, so that it passes CodeIgniter's empty data check.
+
+```php
+$users = model('UserModel');
+$user = new User([
+    'username' => 'foo-bar'
+]);
+$users->save($user);
+
+// Get the updated user so we have the ID...
+$user = $users->findById($users->getInsertID());
+
+// Store the email/password identity for this user.
+$user->createEmailIdentity($this->request->getPost(['email', 'password']));
+
+// Add to default group
+$users->addToDefaultGroup($user);
+```
+
+### Deleting Users
+
+A user's data can be spread over a few different tables so you might be concerned about how to delete all of the user's data from the system. This is handled automatically at the database level for all information that Shield knows about, through the `onCascade` settings of the table's foreign keys. You can delete a user like any other entity.
```

Comment:
This is a bit misleading since our UserModel implements soft deletes by default. Please either update the example to be a hard delete or differentiate the process more.